### ISSUE TITLE
Need to rethink partial test response validation 

### ISSUE THREAD

**Issue by atvassilev** [usnistgov/ACVP#8]
A new issue from NIST Sprint 1:

The validation of partial test results offers some flexibility but creates substantial complications in managing validation status on the NIST server. We believe all benefits intended by this protocol feature can be attained in other ways that result in much simpler and robust validation server architecture.  In order to simplify the storage and data processing on the ACV servers it would be beneficial to disallow the submission of partial test results. We would like to remove this capability from the protocol.

**Comment by atvassilev** [comment]
Fixed with revisions 15234a8 and 2ae0dfa

**Comment by atvassilev** [comment]
Mistakenly closed this. Reopening.

**Comment by atvassilev** [comment]
The latest version of the protocol specification is still ambiguous about handling of partial submissions, meaning it is not specified. I think this may create confusion in actual implementations of the client and server parts of the protocol leading to incompatibilities.  Some text in Section 11.3 is needed to prevent this. Perhaps we can discuss this issue on the call today.

**Comment by bfussell** [comment]
Sure, we can cover that today.

Barry

**Comment by jfigus** [comment]
I agree, a partial upload of an individual vector set should not be allowed.  This would over-complicate the implementation requirements.  But a single test session may have multiple vector sets associated.  Given we're using REST, which is a stateless protocol, we need to retain the concept of uploading partial results across a single test session (e.g. 4 of 10 vector sets have been uploaded, client is still processing the remaining 6 vector sets and will upload when ready).

**Comment by atvassilev** [comment]
John,

Happy New Year, good to hear from you again. I agree we need to allow a single test vector to be submitted when a session contains many, as long as each vector is complete. In fact the NIST team discussed the need to submit one test vector at a time to accommodate large test cases that may grow too big if several test vectors are submitted together. This would mean a client needs to keep track of the test session and submit a response to every contained test vector, one by one. If we all agree, I could submit another PR to add this clarification to Section 11.3.
